### PR TITLE
feat: Change Recipe Owner

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- feature

## What this PR does / why we need it:

_(REQUIRED)_

This PR gives a proper interface for updating a recipe's owner. While editing the recipe:
image

(note the household is there for convenience, so you don't accidentally switch a recipe to another household if you didn't want to)

And on the data management page:
image
image

## Which issue(s) this PR fixes:

_(REQUIRED)_

N/A, it gets asked on Discord a lot

## Special notes for your reviewer:

_(fill-in or delete this section)_

I had to add a generic bulk update route for recipes, similar to other existing bulk update routes.

## Testing

_(fill-in or delete this section)_

Added one pytest test.

### CODE REVIEW

**Comment by andrewdunndev** [comment]
Awesome, this is exactly a feature I was looking for!

File: frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageEditorToolbar.vue [hunk 1]
```diff
@@ -1,28 +1,52 @@
...
+        <template #prepend>
+          <UserAvatar :user-id="recipe.userId" :tooltip="false" />
+        </template>
+      </v-select>
+      <v-card-text v-if="ownerHousehold" class="pa-0 d-flex" style="align-items: flex-end;">
+        <v-spacer />
+        <v-icon>{{ $globals.icons.household }}</v-icon>
+        <span class="pl-1">{{ ownerHousehold.name }}</span>
+      </v-card-text>
+    </v-container>
```
Comment (boc-the-git) [inline frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageEditorToolbar.vue]: Thoughts on limiting this to admins only?

File: frontend/pages/group/data/recipes.vue [hunk 2]
```diff
@@ -314,8 +342,6 @@ export default defineComponent({
 
       const { response, data } = await api.bulk.bulkDelete({ recipes });
```
Comment (boc-the-git) [inline frontend/pages/group/data/recipes.vue]:
```suggestion
      await api.bulk.bulkDelete({ recipes });
```

[sand] unused variables
image

File: frontend/pages/group/data/recipes.vue [hunk 3]
```diff
@@ -337,7 +363,21 @@ export default defineComponent({
 
       const { response, data } = await api.bulk.bulkSetSettings({ recipes, settings: recipeSettings });
```
Comment (boc-the-git) [inline frontend/pages/group/data/recipes.vue]:
```suggestion
      await api.bulk.bulkSetSettings({ recipes, settings: recipeSettings });
```

[sand] unused variables

File: frontend/pages/group/data/recipes.vue [hunk 4]
```diff
@@ -337,7 +363,21 @@ export default defineComponent({
...
+      if(!selected.value.length || !selectedOwner.value) {
+        return;
+      }
+
+      selected.value.forEach((r) => {
+        r.userId = selectedOwner.value;
+      });
+
+      loading.value = true;
+      const { response, data } = await api.recipes.updateMany(selected.value);
```
Comment (boc-the-git) [inline frontend/pages/group/data/recipes.vue]:
```suggestion
      await api.recipes.updateMany(selected.value);
```

[sand] unused vars

File: mealie/routes/recipe/recipe_crud_routes.py [hunk 5]
```diff
@@ -466,6 +468,31 @@ def update_one(self, slug: str, data: Recipe):
...
+            lambda: defaultdict(list)
+        )
+        for recipe in data:
+            r = self.service.update_one(recipe.id, recipe)  # type: ignore
+            updated_by_group_and_household[r.group_id][r.household_id].append(r)
+
+        all_updated: list[Recipe] = []
+        if updated_by_group_and_household:
+            for group_id, household_dict in updated_by_group_and_household.items():
+                for household_id, updated in household_dict.items():
```
Comment (boc-the-git) [inline mealie/routes/recipe/recipe_crud_routes.py]: [sand] I'd probably prefer `updated_recipes` or something, though I do appreciate we're in a recipe specific file

**Review by boc-the-git — DISMISSED** [review]

File: frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageEditorToolbar.vue (hunk 1 above)
Comment (michael-genson) [reply frontend/components/Domain/Recipe/RecipePage/RecipePageParts/RecipePageEditorToolbar.vue]: That's a good idea, we can limit it to admins or the current owner

**Review by boc-the-git — DISMISSED** [review]

File: mealie/routes/recipe/recipe_crud_routes.py (hunk 5 above)
Comment (michael-genson) [reply mealie/routes/recipe/recipe_crud_routes.py]: Yeah I copied this from the cookbooks route. I'll update both

**Comment by michael-genson** [comment]
Should be good now!

**Review by boc-the-git — APPROVED** [review]

**Pull request merged**